backend/apiServer/controller/CodeSnippetController.go

```diff
@@ -107,6 +107,17 @@ func (codeSnippet CodeSnippetController) GetImplementation(w http.ResponseWriter
 			return
 		}
 
+		if startLine < 1 {
+			http.Error(w, "lineStart must be 1 or greater", http.StatusBadRequest)
+			util.TypeLogger.Warn("%s: Requested lineStart was befor start of file: %d", packageName, startLine)
+			return
+		}
+
+		if startLine > endLine {
+			http.Error(w, "Cant give negative interval", http.StatusBadRequest)
```

Comment:
reverse interval instead?

Reply:
I would think that could cause confusion, unless they are parsing their file backwards then the only reason they would give a negative interval is if an error occurred.